realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -62,14 +62,14 @@ public void executeDelayedDisposal() {
             }
             abandonedQueries.clear();
 
-            cleanRows();
+            cleanNativeReferences();
         }
     }
 
-    public void cleanRows() {
+    public void cleanNativeReferences() {
         NativeObjectReference reference = (NativeObjectReference) referenceQueue.poll();
         while (reference != null) {
-            UncheckedRow.nativeClose(reference.nativePointer);
+            reference.cleanup();
```

Comment:
`PhantomeReference` are not cleared automatically compared to Weak/Soft, (because of patent apparently)
we need to call `clear` anyway to GC the PhantomReference itself, I think if you override `clear` to close the pointer as well you can achieve both things